source/MRViewer/MRCommandLoop.cpp

```diff
@@ -94,6 +111,9 @@ void CommandLoop::addCommand_( CommandFunc func, bool blockThread, StartPosition
     getViewerInstance().postEmptyEvent();
     if ( blockThread )
         cmd->callerThreadCV.wait( lock );
+
+    if ( blockThread && exception )
+        std::rethrow_exception( exception );
```

Comment:
but in one if block?
```suggestion
    if ( blockThread )
    {
        cmd->callerThreadCV.wait( lock );
        if ( exception )
            std::rethrow_exception( exception );
    }